consensus/src/consensus_agent/requests.rs

```diff
@@ -106,4 +110,44 @@ impl<B: AbstractBlockchain<'static> + 'static, MA: MessageAdapter<B::Block> + 's
         });
         tokio::spawn(future);
     }
+
+    pub(super) fn on_get_epoch_transactions(&self, get_epoch_transactions_message: GetEpochTransactionsMessage) {
+        trace!("[GET-EPOCH-TRANSACTIONS] from {}", self.peer.peer_address());
+        if !self.state.write().epoch_transactions_limit.note_single() {
+            warn!("Rejecting GetEpochTransactions message - rate-limit exceeded");
+            // TODO: Should we send something?
+            return;
+        }
+
+        let transactions: Option<Vec<Transaction>> = self.blockchain.get_epoch_transactions(get_epoch_transactions_message.epoch, |tx| tx.clone(), None);
+        if transactions.is_none() {
+            debug!("[GET-EPOCH-TRANSACTIONS] Could not determine transactions for epoch {:?}", get_epoch_transactions_message.epoch);
+            return;
+        }
+        let transactions = transactions.unwrap();
```

Comment:
This is unidiomatic. You could use a `if let`. But not important :man_shrugging:

Reply:
You can use the `?` operator here. To still have the `debug!` here, you can use `map_err`.